Automattic/jetpack
# RNMobile: Fix edge case where VideoPress v5 could mistakenly appears empty

Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/6670

## Proposed changes:

* It is valid that the Video block [does not have an ID associated with it](https://github.com/Automattic/jetpack/blob/bacf2f6c4cdd03aac04dde2eac9356c5cc4e544e/projects/plugins/jetpack/extensions/blocks/videopress/edit.native.js#L219) in some cases after uploading a video via a URL. However, those videos can still be valid VideoPress URLs with a `GUID` (as well as other VideoPress metadata). At the moment, the block's [`isSourcePresent`](https://github.com/Automattic/jetpack/blob/e4b7d24996f5d3d0f612681c74f4d7c4d415f19a/projects/plugins/jetpack/extensions/blocks/videopress/edit.native.js#L298) bool returns false in these cases, leading to the block falsely appearing empty.
* To resolve this edge case, the check for a valid ID within `isSourcePresent` has been removed in https://github.com/Automattic/jetpack/pull/35957/commits/8f1a4c3d1aa9fcf75a594919f7f024a7adaaa4a8. It is only necessary for a video to either have a valid `src` or `GUID` to confirm that a source is present.

### Other information:


## Jetpack product discussion

N/A

## Does this pull request change what data or activity we track or use?

No, it does not.

## Testing instructions:

* Copy the following URL: https://videos.files.wordpress.com/qDiVCMnq/0e2809330-1-2.mp4. This URL was obtained by getting the download URL from the VideoPress player.
* Create a post in a Simple site.
* Add a Video block.
* Insert the copied URL.
* Observe that the video can be played within the editor.
* Save the post and open it again.
* Observe that the post is marked as dirty. In this step, the video URL changed to the regular VideoPress URL. **Note, this happens within the shared `save.js` file.**
* Save the post and open it again.
* Verify that the block is _not_ in the empty state.
